apps/src/templates/AchievementDialog.jsx

```diff
@@ -250,11 +262,22 @@ const AchievementDialog = Radium(React.createClass({
     } else {
       feedbackMessage = locale[tooManyBlocks ? 'numBlocksNeeded' : 'nextLevel'](params);
     }
-    const numPoints = 1 +
-      (showNumBlocksRow && !tooManyBlocks ? 1 : 0) +
-      (!tooManyHints ? 1 : 0);
+    const completionPoints = 1;
+    const numBlocksPoints = showNumBlocksRow && !tooManyBlocks ? 3 : 0;
+    const hintsPoints = this.props.hintsUsed === 0 ? 2 : this.props.hintsUsed === 1 ? 1 : 0;
+    const numPoints = completionPoints + numBlocksPoints + hintsPoints;
+
     const dotsUrl = `url(${this.props.assetUrl('media/dialog/dots.png')})`;
 
+    const achievementRows = [
+      this.achievementRow(true /* show */, true /* success */, completionPoints,
+          locale.puzzleCompleted()),
+      this.achievementRow(showNumBlocksRow, !tooManyBlocks, numBlocksPoints,
+          this.blocksUsedMessage(blockDelta, params)),
+      this.achievementRow(true /* show */, !tooManyHints, hintsPoints,
```

Comment:
this is a bit confusing because it implies the hints section will always show, but if we are only using this for Monday, I think it's okay for now.